test_programs/compile_success_empty/int_or_field_casting_loop/src/main.nr

```diff
@@ -0,0 +1,15 @@
+// issue #3639
+fn main() {
+    for i in 0..8 {
+        for j in 0..8 {
+            if  (i as u64 + j as u64 < 8) {
+                // pass
+            }
+        }
+    }
+}
+
+#[test]
+fn test_main() {
+    main();
+}
```

Comment:
```suggestion
```
If you need to test `main` locally, `nargo execute` is usually better unless you want to write several different tests functions

Reply:
This test also works on master